### ISSUE TITLE
adding statements to existing items with batch(new=False) deletes existing statements.

### ISSUE THREAD

**Issue by MHuberFaust** [UB-Mannheim/RaiseWikibase#20]
Hi,

I want to add statements to existing items using batch(new=False) to enrich the items in my Wikibase instance.

Unfortunately, already existing statements were deleted.

Having batch(new=False) add statements without deleting already existing ones would be quite useful. Either as default behaviour or as an option.
I will look into it as soon as I find the time, but that might take a while.

Best,
Michael

**Comment by renat-kaufmann** [comment]
Right, because it replaces the whole JSON representation of an item. To add what you propose we would need to read `old_text` from `text`-table (https://www.mediawiki.org/wiki/Manual:Text_table), to convert it to JSON, to add the statements to it, and only then to write it via `batch(new=False)`.
